### PR TITLE
Customer bug fix

### PR DESCRIPTION
**Pull request by alexeyoverchenko**
some changes in userMapper

### CODE REVIEW

File: src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java [hunk 1]
```diff
@@ -30,6 +31,7 @@ public UserDto map(User user) {
                 .deletedAt(user.getDeletedAt())
                 .addressDto(user.getAddress() != null ? addressMapper.map(user.getAddress()) : null)
                 .warehouseDto(Objects.isNull(user.getWarehouse()) ? null : warehouseMapper.map(user.getWarehouse()))
+                .customerDto(customerMapper.map(user.getCustomer()))
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java]: We need to handle null case like in the line above.

File: src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java [hunk 2]
```diff
@@ -60,6 +62,7 @@ public User map(UserDto userDTO) {
                 .warehouse(Objects.isNull(userDTO.getWarehouseDto())
                         ? null :
                         warehouseMapper.map(userDTO.getWarehouseDto()))
+                .customer(customerMapper.map(userDTO.getCustomerDto()))
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java]: Need to handle null case.

**Review by eugenmolchanov — CHANGES_REQUESTED** [review]
Please, see the comments.

File: src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java (hunk 1 above)
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java]: Optional.ofNullable(user.getSmth()).map(mapper::map).orElse(null)

File: src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java (hunk 2 above)
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java]: Optional.ofNullable(user.getSmth()).map(mapper::map).orElse(null)

File: src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java (hunk 1 above)
Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java]: done

File: src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java (hunk 2 above)
Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java]: done

File: src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java (hunk 1 above)
Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java]: done

File: src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java (hunk 2 above)
Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/dto/mapper/UserMapper.java]: done

**Review by eugenmolchanov — APPROVED** [review]

**Review by svareyko — APPROVED** [review]

**Pull request merged**